client/src/main/kotlin/io/spine/examples/chatspn/TestDataChatProvider.kt

```diff
@@ -0,0 +1,151 @@
+        mutableMapOf()
+    private val vlad = userId("vladId")
+    private val artem = userId("artemId")
+    private val alex = userId("alexId")
+    private val artemChat = chatId("chat1")
+    private val alexChat = chatId("chat2")
+
+    init {
+        val artemChatPreview = ChatPreview
+            .newBuilder()
+            .setPersonalChat(
+                PersonalChatView.newBuilder()
+                    .setCreator(vlad)
+                    .setMember(artem)
+                    .vBuild()
+            )
+            .setId(artemChat)
+            .vBuild()
+        val alexChatPreview = ChatPreview
+            .newBuilder()
+            .setPersonalChat(
+                PersonalChatView
+                    .newBuilder()
+                    .setCreator(alex)
+                    .setMember(vlad)
+                    .vBuild()
+            )
+            .setId(alexChat)
+            .vBuild()
+
+        chatsFlow.value = listOf(artemChatPreview, alexChatPreview)
+
+        chatMessagesFlow[artemChat] = MutableStateFlow(listOf())
+        chatMessagesFlow[alexChat] = MutableStateFlow(listOf())
+    }
+
+    override fun messages(chat: ChatId): StateFlow<List<MessagePreview>> {
+        if (!chatMessagesFlow.containsKey(chat)) {
+            throw IllegalStateException("Chat not found")
+        }
+        return chatMessagesFlow.getOrDefault(chat, MutableStateFlow(mutableListOf()))
+    }
+
+    override fun chats(): StateFlow<List<ChatPreview>> {
+        return chatsFlow
+    }
+
+    override fun sendMessage(chat: ChatId, content: String) {
+        sendMessage(vlad, chat, content)
+    }
+
+    private fun sendMessage(user: UserId, chat: ChatId, content: String) {
+        val message = createMessage(user, content)
+        chatMessagesFlow[chat]?.value = chatMessagesFlow[chat]?.value?.plus(listOf(message))!!
+        val optionalChat = chatsFlow
+            .value
+            .stream()
+            .filter { chatPreview -> chatPreview.id.equals(chat) }
+            .findFirst()
+        if (optionalChat.isPresent) {
+            val chatPreview = optionalChat.get()
+            val newChat = chatPreview
+                .toBuilder()
+                .setLastMessage(message)
+                .vBuild()
+            val chats = chatsFlow.value
+            val newList = chats.subList(0, chats.indexOf(chatPreview)) +
+                    listOf(newChat) +
+                    chats.subList(chats.indexOf(chatPreview) + 1, chats.size)
+            chatsFlow.value = newList
+        }
+    }
+
+    /**
+     * Periodically sends messages to the chat.
+     */
+    suspend fun startChatting() {
+        while (true) {
+            sendMessage(artem, artemChat, "Hello${(Math.random() * 100).toInt()}")
+            delay(1000)
+            sendMessage(alex, alexChat, "Hello${(Math.random() * 100).toInt()}")
+            delay(3000)
+        }
+    }
+
+    private fun userId(id: String): UserId {
```

Comment:
Let's create private extension functions for `String` instead of these Java-like things.